### PR TITLE
UOE-10620: Resolved regex flow for test = 1

### PR DESCRIPTION
**Pull request by pm-priyanka-bagade**

### CODE REVIEW

**Review by AvinashKapre — APPROVED** [review]
LGTM

File: modules/pubmatic/openwrap/bidderparams/pubmatic.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func PreparePubMaticParamsV25(rctx models.RequestCtx, cache cache.Cache, bidRequ
 
 	slots, slotMap, slotMappingInfo, _ := getSlotMeta(rctx, cache, bidRequest, imp, impExt, partnerID)
 
-	if rctx.IsTestRequest > 0 {
+	if rctx.IsTestRequest > 1 {
```
Comment (pm-nilesh-chate) [inline modules/pubmatic/openwrap/bidderparams/pubmatic.go:31]: what about test requests?

Comment (pm-priyanka-bagade) [reply modules/pubmatic/openwrap/bidderparams/pubmatic.go]: changed please check

File: modules/pubmatic/openwrap/bidderparams/pubmatic_test.go [hunk 2]
```diff
@@ -1071,3 +1075,118 @@ func createSlotMapping(slotName string, mappings map[string]interface{}) models.
 		OrderID:      0,
 	}
 }
+
+func TestGetMatchingSlotAndPattern(t *testing.T) {
```
Comment (pm-saurabh-narkhede) [inline modules/pubmatic/openwrap/bidderparams/pubmatic_test.go:1079]: @pm-priyanka-bagade  can we have test cases for test = 1 and test >=2?

File: modules/pubmatic/openwrap/bidderparams/pubmatic.go [hunk 3]
```diff
@@ -45,28 +36,21 @@ func PreparePubMaticParamsV25(rctx models.RequestCtx, cache cache.Cache, bidRequ
...
-				if slotMappingInfo.HashValueMap != nil {
-					if v, ok := slotMappingInfo.HashValueMap[matchedPattern]; ok {
-						extImpPubMatic.AdSlot = v
-						imp.TagID = hash // TODO, make imp pointer. But do other bidders accept hash as TagID?
-					}
-				}
-			}
-
-			break
+	if rctx.IsTestRequest == 1 {
```
Comment (pm-saurabh-narkhede) [inline modules/pubmatic/openwrap/bidderparams/pubmatic.go:39]: can we have test cases for test request =1 and >0?

Comment (pm-priyanka-bagade) [reply modules/pubmatic/openwrap/bidderparams/pubmatic.go:39]: done

**Review by pm-saurabh-narkhede — APPROVED** [review]

**Review by pm-nilesh-chate — APPROVED** [review]

**Pull request merged**